formats/html.ts

```diff
@@ -1,44 +1,339 @@
+				break;
+			case "https:":
+			case "http:":
+				response = await this.requestHTTP(type, url);
+				break;
+			default:
+				throw new Error(url.href);
+		}
+		if (!await this.filterAttachment(response)) {
+			return null;
+		}
+		const { data, extension } = response;
+		let filename = getURLFilename(url);
+		if (extension) {
+			if (splitFilename(filename).extension !== extension) {
+				filename += `.${extension}`;
+			}
+		} else {
+			filename += `.noext.${{
+				"audio": "mp3",
+				"img": "bmp",
+				"video": "mp4",
+			}[type]}`;
+		}
+		return await this.writeAttachment(mdFile, filename, data);
+	}
+
+	async requestFile(type: TypedResponse["type"], url: URL) {
+		return {
+			type,
+			data: (await fsPromises.readFile(nodeUrl.fileURLToPath(url.href))).buffer,
+			extension: splitFilename(getURLFilename(url)).extension,
 		};
+	}
 
-		for (let file of files) {
+	requestHTTP(type: TypedResponse["type"], url: URL) {
+		return this.requestHTTPExecutor.run(async () => {
+			url = new URL(url.href);
+			let response;
 			try {
-				let htmlContent = await file.readText();
-				let mdContent = htmlToMarkdown(htmlContent);
-				await this.saveAsMarkdownFile(folder, file.basename, mdContent);
-				results.total++;
+				url.protocol = "https:";
+				response = await requestURL(url);
 			} catch (e) {
-				console.error(e);
-				results.failed.push(file.toString());
+				try {
+					url.protocol = "http:";
+					response = await requestURL(url);
+				} catch (e2) {
+					console.error(e2);
+					throw e;
+				}
 			}
+			return {
+				type,
+				data: response.data,
+				extension: extension(response.mime) || splitFilename(getURLFilename(url)).extension,
+			};
+		});
+	}
+
+	writeAttachment(mdFile: TFile, filename: string, data: ArrayBufferLike) {
+		return this.writeAttachmentExecutor.run(async () => {
+			const { basename, extension } = splitFilename(sanitizeFileName(filename));
+			// @ts-ignore
+			const path: string = await this.app.vault.getAvailablePathForAttachments(basename, extension, mdFile);
+			return await this.app.vault.createBinary(path, data);
+		});
+	}
+
+	async filterAttachment(response: TypedResponse) {
+		const { data } = response;
+		return this.filterAttachmentSize(data) && await this.filterImageSize(response);
+	}
+
+	filterAttachmentSize(data: ArrayBufferLike) {
+		const { byteLength } = data;
+		return !this.attachmentSizeLimit || byteLength <= this.attachmentSizeLimit;
+	}
+
+	async filterImageSize(response: TypedResponse) {
+		const { data, type } = response;
+		if (!this.minimumImageSize || type !== "img") {
+			return true;
+		}
+		let size;
+		try {
+			size = await imageSize(new Blob([data]));
+		} catch {
+			return true;
+		}
+		const { height, width } = size;
+		return width >= this.minimumImageSize && height >= this.minimumImageSize;
+	}
+}
+
+interface TypedResponse {
+	type: "audio" | "img" | "video";
+	data: ArrayBufferLike;
+	extension: string;
+}
+
+function escapeRegExp(str: string) {
```

Comment:
There is also `escapeStringRegexp` inside `yarle`, but using that causes "Invalid escape" error when creating a new `RegExp`. I don't want to break the yarle importer so I didn't fix and use that function.